homeassistant/components/stream/recorder.py

```diff
@@ -36,9 +36,9 @@ def recorder_save_worker(file_out: str, segments: List[Segment]):
         # Remux video
         for packet in source.demux(source_v):
             if packet is not None and packet.dts is not None:
-                if first_pts is None:
-                    first_pts = packet.pts
-
+                if packet.pts < segment.start_pts[0]:
```

Comment:
Does this just work around when segments[0].start_pts[0] is incorrect?

Reply:
It gets around when `packet.pts` is `0` based instead of `start_pts` based, which seems to happen for the "lookback" segments we pull from the hls provider.

Reply:
Interesting...can't see why the lookback segments would be coming out as 0...will have to take a look with fresh eyes later.

Reply:
The PTS and DTS get set to zero when the container option "empty_moov" is set. I think it's from this line https://ffmpeg.org/doxygen/trunk/movenc_8c_source.html#l05657
The timestamps get reset to match the last timestamps of the existing track, which is empty. This explains why we have to add in the tfdt offsets. From looking at that file, it seems like we actually might be able to skip the box editing if we set "frag_discont" and "fragment_index". I'll play around with it later when I have time.
BTW, any thoughts on how to deal with timestamp overflow/rollover?